chain/client/src/client.rs

```diff
@@ -2004,4 +2012,63 @@ impl Client {
             next_blocks_log.join("\n")
         ))
     }
+
+    pub fn detailed_upcoming_blocks_info_as_web(&self) -> ChunkInfoView {
+        let height_status_map = match self.detailed_upcoming_blocks_info() {
+            Ok(m) => m,
+            Err(_) => HashMap::new(),
+        };
+
+        let next_blocks_log = height_status_map
+            .keys()
+            .sorted()
+            .map(|height| {
+                let val = height_status_map.get(height).unwrap();
+
+                let block_debug = val
+                    .iter()
+                    .map(|entry| {
+                        let block_info = entry.1;
+                        let chunk_status = block_info
+                            .chunk_hashes
+                            .iter()
+                            .map(|it| {
+                                if block_info.chunks_completed.contains(it) {
+                                    "(done)"
+                                } else if block_info.chunks_received.contains(it) {
+                                    "(received)"
+                                } else if block_info.chunks_requested.contains(it) {
+                                    "(requested)"
+                                } else {
+                                    "."
+                                }
+                            })
+                            .collect::<Vec<&str>>();
+
+                        let in_progress_str = match block_info.in_progress_for {
+                            Some(duration) => format!("in progress for: {:?}", duration),
+                            None => "".to_string(),
+                        };
+                        let in_orphan_str = match block_info.in_orphan_for {
+                            Some(duration) => format!("orphan for {:?}", duration),
+                            None => "".to_string(),
+                        };
+
+                        format!(
```

Comment:
how well does it look on the webpage?

(as this kind of format was originally written for the console)

Reply:
Screenshot coming soon.